### ISSUE TITLE
Weird escaping necessary for schemas in PostgreSQL

### ISSUE THREAD

**Issue by einfallstoll** [PHPixie/Database#1]
Assume you want to query `mytable` of the following database structure:

```
mydatabase
   ⌙-> myschema
          ⌙-> mytable
```

A general query would look like this (given that you selected the database):

```
SELECT * FROM "myschema"."mytable";
```

Translating it to the `Database` component:

```
$gateways = $query
    ->table('myschema.mytable');

/*
execute():

Fatal error: Uncaught exception 'PDOException' with message 'SQLSTATE[42P01]: Undefined table: 7 ERROR: relation "myschema.mytable" does not exist LINE 1: SELECT * FROM "myschema.mytable" ^' in ...

parse():

object(PHPixie\Database\Type\SQL\Expression)#51 (2) { ["sql"]=> string(28) "SELECT * FROM "myschema.mytable"" ["params"]=> array(0) { } }

SQL Query:
SELECT * FROM "myschema.mytable" // invalid
*/

$gateways = $query
    ->table('"myschema"."mytable"');

/*
execute():

Fatal error: Uncaught exception 'PDOException' with message 'SQLSTATE[42601]: Syntax error: 7 ERROR: zero-length delimited identifier at or near """" LINE 1: SELECT * FROM ""myschema"."mytable"" ^' in 

parse():

object(PHPixie\Database\Type\SQL\Expression)#51 (2) { ["sql"]=> string(32) "SELECT * FROM ""myschema"."mytable""" ["params"]=> array(0) { } }

SQL Query:
SELECT * FROM ""myschema"."mytable"" // invalid
*/
```

For a correct escaping I have to write the table like this: `myschema"."mytable` because the component just adds a double quote at the beginning and at the end of the table definition.

```
$query = $connection->selectQuery();
$gateways = $query
    ->table('myschema"."mytable');

parse():

object(PHPixie\Database\Type\SQL\Expression)#51 (2) { ["sql"]=> string(30) "SELECT * FROM "myschema"."mytable"" ["params"]=> array(0) { } }

SQL Query:
SELECT * FROM "myschema"."mytable" // valid
```

I suggest to escape the table like this:
- `explode` table by `.` (dot)
- escape each component
- `implode` components by `.` (dot)

This would also work for MySQL, where you can query by mydatabase.mytable

**Comment by dracony** [comment]
Good idea. Thanks. Will implement this today

**Comment by dracony** [comment]
Fixed in 3.1

**Comment by einfallstoll** [comment]
Works as expected. Thank you.
